### PR TITLE
add global min and max threshold filters for features

### PR DESCRIPTION
**Pull request by saupchurch**
This add filters for the feature axis to do some global content filtering.  Setting a min value will remove features that have any value less than or equal to the value in any sample in the matrix.  Setting a max will remove any features with a value greater than the parameter in any sample in the matrix.

Using both will result in a matrix with only features that have values in the interval (min, max] for every sample.

A specific use case is using a min value of 0 to result in a matrix with all the zeros removed.

### CODE REVIEW

**Comment by saupchurch** [comment]
Following discussion we are improving clarity by changing the behavior of the threshold values to include both the min and max values - i.e. the closed interval [min, max] instead of the half open interval.  The use case of removing zeros can be accomplished by either supplying some low min value of can be done client side by existing matrix tools.

**Review by hitz — APPROVED** [review]

File: rnaget-openapi.yaml [hunk 1]
```diff
@@ -1406,7 +1406,7 @@ components:
     featureMinParam:
       name: feature_min_value
       in: query
-      description: Sets a minimum expression value for features.  If set the resulting matrix will not contain any features with a value less than or equal to the provided threshold in any sample in the matrix.  The resulting matrix should have values > the supplied threshold in every cell.  For example, specifying a feature_min_value of 0 will remove zeros from the matrix by removing any feature that has a zero value in 1 or more samples.  Using this with feature_max_value will result in matrix values in the half open interval (feature_min_value, feature_max_value].
+      description: Sets a minimum expression value for features.  If set the resulting matrix will not contain any features with a value less than the provided threshold in any sample in the matrix.  The resulting matrix should have values >= the supplied threshold in every cell.  Using this with feature_max_value will result in matrix values in the half closed interval [feature_min_value, feature_max_value].
```
Comment (emi80) [inline rnaget-openapi.yaml:1409]: the word `half` should be removed
```suggestion
      description: Sets a minimum expression value for features.  If set the resulting matrix will not contain any features with a value less than the provided threshold in any sample in the matrix.  The resulting matrix should have values >= the supplied threshold in every cell.  Using this with feature_max_value will result in matrix values in the closed interval [feature_min_value, feature_max_value].
```

File: rnaget-openapi.yaml [hunk 2]
```diff
@@ -1415,7 +1415,7 @@ components:
     featureMaxParam:
       in: query
       name: feature_max_value
-      description: Sets a maximum expression value for features.  If set the resulting matrix will not contain any features with a value greater than the provided threshold in any sample in the matrix.  The resulting matrix should have values <= the supplied threshold in every cell.  Using this with feature_min_value will result in matrix values in the half open interval (feature_min_value, feature_max_value].
+      description: Sets a maximum expression value for features.  If set the resulting matrix will not contain any features with a value greater than the provided threshold in any sample in the matrix.  The resulting matrix should have values <= the supplied threshold in every cell.  Using this with feature_min_value will result in matrix values in the closedinterval [feature_min_value, feature_max_value].
```
Comment (emi80) [inline rnaget-openapi.yaml:1418]: missing space
```suggestion
      description: Sets a maximum expression value for features.  If set the resulting matrix will not contain any features with a value greater than the provided threshold in any sample in the matrix.  The resulting matrix should have values <= the supplied threshold in every cell.  Using this with feature_min_value will result in matrix values in the closed interval [feature_min_value, feature_max_value].
```

**Review by emi80 — APPROVED** [review]

File: rnaget-openapi.yaml [hunk 3]
```diff
@@ -1439,6 +1447,24 @@ components:
         type: integer
         format: int32
         minimum: 0
+    featureMinParam:
+      name: feature_min_value
+      in: query
+      description: Sets a minimum expression value for features.  If set the resulting matrix will not contain any features with a value less than the provided threshold in any sample in the matrix.  The resulting matrix should have values >= the supplied threshold in every cell.  Using this with feature_max_value will result in matrix values in the closed interval [feature_min_value, feature_max_value].
```
Comment (emi80) [inline rnaget-openapi.yaml:1453]:
```suggestion
      description: Sets a minimum expression value for features. If set the resulting matrix will not contain any features with a value less than the provided threshold in any sample in the matrix. The resulting matrix should have values >= the supplied threshold in every cell. Using this with feature_max_value will result in matrix values in the closed interval [feature_min_value, feature_max_value].
```

File: rnaget-openapi.yaml [hunk 4]
```diff
@@ -1439,6 +1447,24 @@ components:
...
+      required: false
+      schema:
+        type: number
+        format: float
+        minimum: 0.0
+    featureMaxParam:
+      in: query
+      name: feature_max_value
+      description: Sets a maximum expression value for features.  If set the resulting matrix will not contain any features with a value greater than the provided threshold in any sample in the matrix.  The resulting matrix should have values <= the supplied threshold in every cell.  Using this with feature_min_value will result in matrix values in the closed interval [feature_min_value, feature_max_value].
```
Comment (emi80) [inline rnaget-openapi.yaml:1462]:
```suggestion
      description: Sets a maximum expression value for features. If set the resulting matrix will not contain any features with a value greater than the provided threshold in any sample in the matrix. The resulting matrix should have values <= the supplied threshold in every cell. Using this with feature_min_value will result in matrix values in the closed interval [feature_min_value, feature_max_value].
```

**Review by emi80 — APPROVED** [review]

**Pull request merged**